star/star.go

```diff
@@ -21,6 +21,7 @@ type Repository interface {
 	Create(ctx context.Context, userID string, assetID string) (string, error)
 	GetStargazers(ctx context.Context, cfg Config, assetID string) ([]user.User, error)
 	GetAllAssetsByUserID(ctx context.Context, cfg Config, userID string) ([]asset.Asset, error)
+	GetAllAssetsByUserEmail(ctx context.Context, cfg Config, userID string) ([]asset.Asset, error)
```

Comment:
```go
type filter struct {
    id    string
    email string
}

type Repository interface {
    ...
	GetAllAssets(ctx context.Context, cfg Config, filter filter) ([]asset.Asset, error)
    ...
}
```

have we looked into this^ approach?

Reply:
These 2 have different sql query,
`GetAllAssetsByUserEmail` joined 3 tables
`GetAllAssetsByUserID` joined 2 tables

I am afraid it will just make it more complex